### ISSUE TITLE
authentication and accounts

### ISSUE THREAD

**Issue by xjarlie** [xjarlie/chat#5]
add full account system - auth, dms, private rooms

**Comment by leonskij** [comment]
This could be a good place to check out https://magic.link/ to make the auth easier for you. Just a quick suggestion.

**Comment by xjarlie** [comment]
@leonskidev Yeah that's a really good idea, I'll look into implementing it, thanks!
